### PR TITLE
Remove dependency of format JSON  file from records download

### PR DESCRIPTION
**Pull request by divyav-aot**

### CODE REVIEW

File: forms-flow-web/src/components/FOI/customComponents/Records/index.js [hunk 1]
```diff
@@ -307,315 +392,452 @@ export const RecordsLog = ({
...
+      requestnumber: requestNumber,
+      bcgovcode: bcgovcode,
+      attributes: [],
     };
 
     //remove duplicate records(except duplicate attachments)
     const deduplicatedRecords = removeDuplicateFiles(recordList);
     //get only relevent fields
-    const updatedrecords = getUpdatedRecords(deduplicatedRecords, conversionFormats);
+    const updatedrecords = getUpdatedRecords(deduplicatedRecords);
```
Comment (divyav-aot) [inline forms-flow-web/src/components/FOI/customComponents/Records/index.js:801]: removed "conversionFormats" param from getUpdatedRecords()

File: forms-flow-web/src/components/FOI/customComponents/Records/index.js [hunk 2]
```diff
@@ -633,46 +855,63 @@ export const RecordsLog = ({
...
+      "Exporting files (" + completed + "/" + exporting.length + ")"
+    );
     try {
       for (let record of exporting) {
-        var filepath = record.s3uripath
-        var filename = record.filename
-        if (record.isredactionready && conversionFormats.includes(record.attributes?.extension?.toLowerCase())) {
+        var filepath = record.s3uripath;
+        var filename = record.filename;
+        if (record.isredactionready && record.isconverted) {
```
Comment (divyav-aot) [inline forms-flow-web/src/components/FOI/customComponents/Records/index.js:878]: removed conversionFormats.includes(record.attributes?.extension?.toLowerCase())) and replaced with record.isconverted

File: forms-flow-web/src/components/FOI/customComponents/Records/index.js [hunk 3]
```diff
@@ -1057,63 +1405,76 @@ export const RecordsLog = ({
...
+                  {recordsDownloadList.map((item, index) => {
+                    if (item.id != 0) {
+                      return (
                         <MenuItem
                           className="download-menu-item"
                           key={item.id}
                           value={index}
-                          disabled={item.disabled || conversionFormats?.length < 1}
-                          sx={{ display: 'flex' }}
+                          disabled={item.disabled}
```
Comment (divyav-aot) [inline forms-flow-web/src/components/FOI/customComponents/Records/index.js:1444]: removed conversionFormats?.length < 1 as download is not dependent on conversionFormats.

File: forms-flow-web/src/components/FOI/customComponents/Records/index.js [hunk 4]
```diff
@@ -1550,437 +2078,532 @@ export const RecordsLog = ({
...
-          {record.originalfile!=''  && record.originalfile!=undefined  && <MenuItem
-            onClick={() => {
-                handleDownloadoriginal();
-                setPopoverOpen(false);
-            }}
-          >
-            Download Original
-          </MenuItem>
+    const ActionsPopover = ({ RestrictViewInBrowser, record }) => {
+      return (
```
Comment (divyav-aot) [inline forms-flow-web/src/components/FOI/customComponents/Records/index.js:2473]: const conversionFormats = useSelector((state) => state.foiRequests.conversionFormats)
removed above code as it is not needed any more

File: forms-flow-web/src/components/FOI/customComponents/Records/index.js [hunk 5]
```diff
@@ -1550,437 +2078,532 @@ export const RecordsLog = ({
...
+              <MenuItem
+                onClick={() => {
+                  handleDownloadoriginal();
+                  setPopoverOpen(false);
+                }}
+              >
+                Download Original
+              </MenuItem>
+            )}
+            {((record.isredactionready && record.isconverted) ||
```
Comment (divyav-aot) [inline forms-flow-web/src/components/FOI/customComponents/Records/index.js:2537]: replaced conversionFormats.includes(record.attributes?.extension?.toLowerCase()) with record.isconverted

File: forms-flow-web/src/components/FOI/customComponents/Records/util.js [hunk 6]
```diff
@@ -1,142 +1,162 @@
...
-  
-    // Create a map of attachments based on parentid
-    for (const attachment of attachments) {
-      const parentid = attachment.parentid;
-      if (!attachmentsMap[parentid]) {
-        attachmentsMap[parentid] = [];
-      }
-      attachmentsMap[parentid].push(attachment);
+
+export const getPDFFilePath = (item) => {
```
Comment (divyav-aot) [inline forms-flow-web/src/components/FOI/customComponents/Records/util.js:17]: removed conversionFormats param from getPDFFilePath()

File: forms-flow-web/src/components/FOI/customComponents/Records/util.js [hunk 7]
```diff
@@ -1,142 +1,162 @@
...
+  let pdffilepath = item.s3uripath;
+  let pdffilename = item.filename;
+
+  if (
+    (item.isredactionready && item.isconverted) ||
```
Comment (divyav-aot) [inline forms-flow-web/src/components/FOI/customComponents/Records/util.js:22]: replaced conversionFormats.includes(item.attributes?.extension?.toLowerCase()) with record.isconverted

File: forms-flow-web/src/components/FOI/customComponents/Records/util.js [hunk 8]
```diff
@@ -1,142 +1,162 @@
...
+    (item.attributes?.isattachment &&
+      item.attributes?.trigger === "recordreplace")
+  ) {
+    pdffilepath = pdffilepath.substr(0, pdffilepath.lastIndexOf(".")) + ".pdf";
+    pdffilename += ".pdf";
+  }
+  return [pdffilepath, pdffilename];
+};
+
+function arrangeAttachments(attachments, parentDocumentMasterId) {
```
Comment (divyav-aot) [inline forms-flow-web/src/components/FOI/customComponents/Records/util.js:32]: removed conversionFormats from arrangeAttachments()

File: forms-flow-web/src/components/FOI/customComponents/Records/util.js [hunk 9]
```diff
@@ -1,142 +1,162 @@
...
-         return total + +_file.filesize;
-       }, 0);
- }
- 
- // calculate final file size
+
+  // Start arranging attachments from the root level
+  arrangeChildren(parentDocumentMasterId);
+  getUpdatedRecords(arrangedAttachments, true);
+  return getUpdatedRecords(arrangedAttachments, true);
```
Comment (divyav-aot) [inline forms-flow-web/src/components/FOI/customComponents/Records/util.js:59]: removed conversionFormats param from getUpdatedRecords()

File: forms-flow-web/src/components/FOI/customComponents/Records/util.js [hunk 10]
```diff
@@ -1,142 +1,162 @@
...
+}
+
+// Get records with only necessary fields
+export const getUpdatedRecords = (_records, isattachment = false) => {
```
Comment (divyav-aot) [inline forms-flow-web/src/components/FOI/customComponents/Records/util.js:63]: removed conversionFormats param from getUpdatedRecords()

File: forms-flow-web/src/components/FOI/customComponents/Records/util.js [hunk 11]
```diff
@@ -1,142 +1,162 @@
...
+  return _records.map((_record) => {
+    const [filepath, filename] = getPDFFilePath(_record);
```
Comment (divyav-aot) [inline forms-flow-web/src/components/FOI/customComponents/Records/util.js:65]: removed conversionFormats param from getPDFFilePath()

File: forms-flow-web/src/components/FOI/customComponents/Records/util.js [hunk 12]
```diff
@@ -1,142 +1,162 @@
...
+      filename: filename,
+      s3uripath: filepath,
+      filesize:
+        _record.attributes.convertedfilesize || _record.attributes.filesize,
+      lastmodified: _record.attributes.lastmodified,
+      isduplicate: _record.isduplicate,
+      divisions: _record.attributes.divisions,
+      divisionids: _record.attributes.divisions.map((d) => d.divisionid),
+      attachments: !isattachment
+        ? arrangeAttachments(sortedAttachments, _record.documentmasterid)
```
Comment (divyav-aot) [inline forms-flow-web/src/components/FOI/customComponents/Records/util.js:84]: removed conversionFormats param from arrangeAttachments()

File: request-management-api/request_api/services/records/recordservicegetter.py [hunk 13]
```diff
@@ -150,48 +208,61 @@ def __getcomputingresponse(self, response, filterby, data: any):
...
+            if entry["parentid"] not in [None, ""] and int(entry["parentid"]) == int(
+                data
+            ):
                 result.append(entry)
             else:
                 filtered.append(entry)
-        return filtered, result  
+        return filtered, result
+
+    def __getisconverted(self, _computingresponse):
```
Comment (divyav-aot) [inline request-management-api/request_api/services/records/recordservicegetter.py:227]: Added __getisconverted() to get the isconverted value

File: request-management-api/request_api/services/records/recordservicegetter.py [hunk 14]
```diff
@@ -9,120 +8,174 @@
...
+                _attachement["isattachment"] = True
+                _attachement["s3uripath"] = attachment["filepath"]
+                _attachement["rootparentid"] = record["recordid"]
+                _attachement["rootdocumentmasterid"] = record["documentmasterid"]
+                _attachement["createdby"] = record["createdby"]
+                _attachement["isselected"] = False
+                _attachement["attributes"] = self.__formatrecordattributes(
+                    attachment["attributes"], divisions
+                )
+                _attachement["isconverted"] = self.__getisconverted(attachment)
```
Comment (divyav-aot) [inline request-management-api/request_api/services/records/recordservicegetter.py:98]: get isconverted value

File: request-management-api/request_api/services/records/recordservicegetter.py [hunk 15]
```diff
@@ -9,120 +8,174 @@
...
+            _record["isduplicate"] = _computingresponse["isduplicate"]
+            _record["attributes"] = self.__formatrecordattributes(
+                _computingresponse["attributes"], divisions
+            )
+            _record["isredactionready"] = _computingresponse["isredactionready"]
+            _record["trigger"] = _computingresponse["trigger"]
+            _record["documentmasterid"] = _computingresponse["documentmasterid"]
+            _record["outputdocumentmasterid"] = documentmasterid
+            _record["isselected"] = False
+            _record["isconverted"] = self.__getisconverted(_computingresponse)
```
Comment (divyav-aot) [inline request-management-api/request_api/services/records/recordservicegetter.py:79]: get is converted value

**Review by nkan-aot — APPROVED** [review]

**Pull request merged**